src/tools/wasm-merge.cpp

```diff
@@ -235,6 +235,19 @@ void updateNames(Module& wasm, KindNameUpdates& kindNameUpdates) {
     }
 
   private:
+    Name& resolveName(NameUpdates updates, Name& name) {
```

Comment:
Looks like this takes name by reference - is that to memoize to make later lookups faster?

This code operates in parallel, however, so it is unsafe to modify the data structure. In fact it should probably be declared as const.

Reply:
It's just that I took model on function `mapName` which is taking a name by reference. (I now see it's because it updates the name in place.)
I'm not a C++ expert...